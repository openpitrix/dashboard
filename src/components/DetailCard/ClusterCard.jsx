import React, { Component } from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import { Link } from 'react-router-dom';
import { translate } from 'react-i18next';

import Status from 'components/Status';
import { ProviderName } from 'components/TdName';
import TimeShow from 'components/TimeShow';
import CopyId from './CopyId';

import styles from './index.scss';

@translate()
export default class ClusterCard extends Component {
  static propTypes = {
    appName: PropTypes.string,
    detail: PropTypes.object.isRequired,
    provider: PropTypes.string,
    runtimeName: PropTypes.string,
    userName: PropTypes.string
  };

  static defaultProps = {
    detail: {}
  };

  render() {
    const {
      detail, appName, runtimeName, provider, userName, t
    } = this.props;

    return (
      <div className={styles.detailCard}>
        <div className={classnames(styles.title, styles.noImg)}>
          <div className={styles.name}>{detail.name || t('None')}</div>
          <CopyId id={detail.cluster_id} />
        </div>
        <ul className={styles.detail}>
          {detail.frontgate_id && (
            <li>
              <span className={styles.name}>{t('Frontgate ID')}</span>
              <span className={styles.id}>{detail.frontgate_id}</span>
            </li>
          )}
          <li>
            <span className={styles.name}>{t('Status')}</span>
            <Status
              type={detail.status}
              transition={detail.transition_status}
            />
          </li>
          <li className={styles.setHeight}>
            <span className={styles.name}>{t('Runtime')}</span>
            <Link
              className={styles.value}
              to={`/dashboard/runtime/${detail.runtime_id}`}
            >
              <ProviderName provider={provider} name={runtimeName} />
            </Link>
          </li>
          <li>
            <span className={styles.name}>{t('App')}</span>
            <Link to={`/dashboard/app/${detail.app_id}`}>
              {appName || t('None')}
            </Link>
          </li>
          <li>
            <span className={styles.name}>{t('User')}</span>
            {userName}
          </li>
          <li>
            <span className={styles.name}>{t('Node Count')}</span>
            {(detail.cluster_node_set && detail.cluster_node_set.length) || 0}
          </li>
          <li>
            <span className={styles.name}>{t('Date Updated')}</span>
            <TimeShow time={detail.upgrade_time} type="detailTime" />
          </li>
        </ul>
      </div>
    );
  }
}
