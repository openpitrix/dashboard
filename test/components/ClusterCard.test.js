import { Provider } from 'mobx-react';
import { BrowserRouter } from 'react-router-dom';
import { observable } from 'mobx';

import ClusterCard from 'components/DetailCard/ClusterCard';

describe('Component ClusterCard ', () => {
  let store;

  beforeAll(() => {
    store = observable.object({});
  });

  it('basic render', () => {
    const wrapper = render(
      <Provider rootStore={store}>
        <BrowserRouter>
          <ClusterCard
            detail={{
              name: 'test cluster',
              cluster_id: 'cl_xxxxxx',
              status: 'active',
              owner: 'system',
              create_time: '2019-04-09T03:39:28Z',
              upgrade_time: '2019-04-19T03:39:28Z'
            }}
          />
        </BrowserRouter>
      </Provider>
    );

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
