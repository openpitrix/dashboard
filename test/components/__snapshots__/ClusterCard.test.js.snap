// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component ClusterCard  basic render 1`] = `
<div
  class="detailCard"
>
  <div
    class="title noImg"
  >
    <div
      class="name"
    >
      test cluster
    </div>
    <div
      class="copyOuter"
    >
      id: cl_xxxxxx
      <span
        class="copyId"
        data-clipboard-text="cl_xxxxxx"
      >
        <span
          class="icon icon-copy"
          style="width:16px;height:16px"
        >
          <svg
            class="qicon qicon-copy qicon-dark"
          >
            <use
              href="#qui-copy"
            />
          </svg>
        </span>
      </span>
    </div>
  </div>
  <ul
    class="detail"
  >
    <li>
      <span
        class="name"
      >
        Status
      </span>
      <span
        class="status"
      >
        <i
          class="icon active"
        />
        <span
          class="name"
        >
          Active
        </span>
      </span>
    </li>
    <li>
      <span
        class="name"
      >
        App
      </span>
      <label
        class="appShow"
      >
        <label
          class="appImage"
        >
          <img
            class="img"
            data-origin-url=""
            src=""
          />
        </label>
        <a
          class="appName"
          href="/user/apps/undefined"
        />
      </label>
    </li>
    <li>
      <span
        class="name"
      >
        Version
      </span>
      <label
        class="value"
      >
        <div
          class="versionType"
        />
        <span
          class="versionName"
        />
      </label>
    </li>
    <li
      class="setHeight"
    >
      <span
        class="name"
      >
        Runtime
      </span>
      <div
        class="repoName value"
        title=""
      >
        <span
          class="icon icon- icon"
          style="width:20px;height:20px"
        >
          <svg
            class="qicon qicon- qicon-dark"
          >
            <use
              href="#qui-"
            />
          </svg>
        </span>
      </div>
    </li>
    <li>
      <span
        class="name"
      >
        Creator
      </span>
      <div
        class="userShow value"
      >
        <label
          class="name"
        >
          system
        </label>
      </div>
    </li>
    <li>
      <span
        class="name"
      >
        Create Time
      </span>
      <div
        class="detailTime"
      >
        2019/04/09 11:39:28
      </div>
    </li>
    <li>
      <span
        class="name"
      >
        Date Updated
      </span>
      <div
        class="detailTime"
      >
        2019/04/19 11:39:28
      </div>
    </li>
  </ul>
</div>
`;
